Scripts/detect_public_api_changes/compare.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+# -------------------------------------------------------------------
+# Setup & Info
+# -------------------------------------------------------------------
+
+BRANCH=$1
+REPO=$2
+TARGET="x86_64-apple-ios17.4-simulator"
+SDK="`xcrun --sdk iphonesimulator --show-sdk-path`"
+COMPARISON_VERSION_DIR_NAME="comparison_version_$BRANCH"
+DERIVED_DATA_PATH=".build"
+SDK_DUMP_INPUT_PATH="$DERIVED_DATA_PATH/Build/Products/Debug-iphonesimulator"
+ALL_TARGETS_LIBRARY_NAME="AdyenAllTargets"
+MODULE_NAMES=($(./Scripts/detect_public_api_changes/package_file_helper.swift Package.swift -print-available-targets))
+
+# -------------------------------------------------------------------
+# Convenience Functions
+# -------------------------------------------------------------------
+
+# Echos the entered information for validation
+function printRunInfo() {
+
+    echo "Branch: " $BRANCH
+    echo "Repo:   " $REPO
+    echo "Target: " $TARGET
+    echo "Dir:    " $COMPARISON_VERSION_DIR_NAME
+
+    echo "Modules:"
+    for MODULE in ${MODULE_NAMES[@]}; do
+        echo "-" $MODULE
+    done
+}
+
+# Clones the comparison project into a custom directory
+function setupComparisonRepo() {
+
+    rm -rf $COMPARISON_VERSION_DIR_NAME
+    mkdir $COMPARISON_VERSION_DIR_NAME
+    cd $COMPARISON_VERSION_DIR_NAME
+    git clone -b $BRANCH $REPO
+    cd ..
+}
+
+# Builds the project and temporarily modifies
+# some files to optimize the process
+#
+# $1 - The path to the current project dir
+#
+# Examples
+#
+#  buildProject . # If we're in the current project dir
+#  buildProject ../.. # If we're in the comparison project dir
+function buildProject() {
+
+    # Removing derived data if available
+    rm -rf .build
+
+    # We have to obscure the project file so `xcodebuild` uses the Package.swift to build the module
+    mv Adyen.xcodeproj Adyen.xcode_proj
```

Comment:
I now think maybe it's better to use underscore in the eof? Wouldn't it be more natural?

Reply:
Yes, agree - in the [swift command-line-tool version](https://github.com/Adyen/adyen-ios/pull/1718) this workaround is not needed anymore so would keep it for now